app/src/main/java/in/co/rajkumaar/amritarepo/aums/activities/LoginActivity.java

```diff
@@ -107,19 +126,24 @@ public boolean onTouch(View v, MotionEvent event) {
         } catch (NullPointerException e) {
             e.printStackTrace();
         }
-
         actionDoneCloseInput();
 
+        final Encryption finalEnc = enc;
         login.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 Utils.hideKeyboard(LoginActivity.this);
-                SharedPreferences pref = getSharedPreferences("user", Context.MODE_PRIVATE);
                 SharedPreferences.Editor ed = pref.edit();
                 if (remember.isChecked()) {
-                    ed.putString("username", username.getText().toString());
-                    ed.putString("password", password.getText().toString());
-                    ed.apply();
+                    try {
+                        String encName = finalEnc.encrypt(username.getText().toString().getBytes(StandardCharsets.UTF_8));
```

Comment:
```suggestion
if(finalEnc){
                        String encName = finalEnc.encrypt(username.getText().toString().getBytes(StandardCharsets.UTF_8));
```

Reply:
if finalEnc is null, it'll go to catch block and resume normal execution. Should I change it?

Reply:
Yeah that’s cool then.